src/core/qgsmaprendererjob.h

```diff
@@ -105,6 +106,16 @@ class CORE_EXPORT QgsMapRendererJob : public QObject
     //! Get pointer to internal labeling engine (in order to get access to the results)
     virtual QgsLabelingResults* takeLabelingResults() = 0;
 
+    //! @note Added in QGIS 3.0
+    //! Set the feature filter provider used by the QgsRenderContext of
+    //! each LayerRenderJob.
+    void setFeatureFilterProvider( const QgsFeatureFilterProvider *f ) { mFeatureFilterProvider = f; };
```

Comment:
Would be good to note in the docs about the lifetime of the provider here - something like `ownership is not transferred and the provider must not be deleted before the render job.`